Cannot find module `scroll-into-view-if-needed`

After upgrading to v5.0.5, this error popped up:
`Cannot find module 'scroll-into-view-if-needed' imported from 'pathToWorkspace/node_modules/.pnpm/svelte-multiselect@5.0.5/node_modules/svelte-multiselect/MultiSelect.svelte'`

Not sure if this only happens in pnpm. I need to install `scroll-into-view-if-needed` manually to resolve it.

--- a/package.json
+++ b/package.json
@@ -35,7 +35,6 @@
     "prettier-plugin-svelte": "^2.7.0",
     "rehype-autolink-headings": "^6.1.1",
     "rehype-slug": "^5.0.1",
-    "scroll-into-view-if-needed": "^2.2.29",
     "svelte": "^3.49.0",
     "svelte-check": "^2.8.0",
     "svelte-github-corner": "^0.1.0",

--- a/src/lib/MultiSelect.svelte
+++ b/src/lib/MultiSelect.svelte
@@ -244,7 +244,10 @@
         // around start/end of option list. Find a better solution than waiting 10 ms to.
         setTimeout(() => {
           const li = document.querySelector(`ul.options > li.active`)
-          if (li) li.scrollIntoViewIfNeeded()
+          if (li) {
+            li.parentNode?.scrollIntoView({ block: `center` })
+            li.scrollIntoViewIfNeeded()
+          }
         }, 10)
       }
     }

--- a/src/lib/MultiSelect.svelte
+++ b/src/lib/MultiSelect.svelte
@@ -1,5 +1,4 @@
 <script lang="ts">
-  import scrollIntoView from 'scroll-into-view-if-needed'
   import { createEventDispatcher } from 'svelte'
   import type { DispatchEvents, MultiSelectEvents, Option } from './'
   import { get_label, get_value } from './'
@@ -245,7 +244,7 @@
         // around start/end of option list. Find a better solution than waiting 10 ms to.
         setTimeout(() => {
           const li = document.querySelector(`ul.options > li.active`)
-          if (li) scrollIntoView(li, { scrollMode: `if-needed` })
+          if (li) li.scrollIntoViewIfNeeded()
         }, 10)
       }
     }

--- a/src/lib/index.ts
+++ b/src/lib/index.ts
@@ -36,3 +36,27 @@ export const get_label = (op: Option) => (op instanceof Object ? op.label : op)
 // fallback on label if option is object and value is undefined
 export const get_value = (op: Option) =>
   op instanceof Object ? op.value ?? op.label : op
+
+// Firefox lacks support for scrollIntoViewIfNeeded, see
+// https://github.com/janosh/svelte-multiselect/issues/87
+// this polyfill was copied from
+// https://github.com/nuxodin/lazyfill/blob/a8e63/polyfills/Element/prototype/scrollIntoViewIfNeeded.js
+if (
+  typeof Element !== `undefined` &&
+  !Element.prototype?.scrollIntoViewIfNeeded
+) {
+  Element.prototype.scrollIntoViewIfNeeded = function (centerIfNeeded = true) {
+    const el = this as Element
+    new IntersectionObserver(function ([entry]) {
+      const ratio = entry.intersectionRatio
+      if (ratio < 1) {
+        const place = ratio <= 0 && centerIfNeeded ? `center` : `nearest`
+        el.scrollIntoView({
+          block: place,
+          inline: place,
+        })
+      }
+      this.disconnect()
+    }).observe(this)
+  }
+}

--- a/tsconfig.json
+++ b/tsconfig.json
@@ -1,4 +1,3 @@
-// https://github.com/sveltejs/language-tools/blob/master/docs/preprocessors/typescript.md
 {
   "extends": "./.svelte-kit/tsconfig.json",
   "compilerOptions": {

--- a/vite.config.ts
+++ b/vite.config.ts
@@ -5,9 +5,6 @@ export default {
   plugins: [sveltekit()],
   test: {
     environment: `jsdom`,
-    deps: {
-      inline: [`compute-scroll-into-view`],
-    },
   },
   resolve: {
     alias: {

